framework/src/governance/lib/datazone/resources/datazone-gsr-msk-datasource/index.mjs

```diff
@@ -0,0 +1,280 @@
+        } catch (error) {
+          if (error.$metadata?.httpStatusCode === 404) {
+            // If asset does not exist, it may have been deleted, so remove its parameter
+            console.log(`Asset ${schemaName} not found in DataZone, removing parameter.`);
+            await ssmClient.send(new DeleteParameterCommand({ Name: parameterName }));
+            assetMap.delete(schemaName);
+          } else {
+            console.error('Error creating asset revision:', error);
+          }
+        }
+      } else {
+        // Create new asset and store its ID
+        console.log(`Creating new asset ${schemaName}.`);
+        try {
+          const createResponse = await dataZoneClient.send(new CreateAssetCommand({
+            domainIdentifier: domainId,
+            owningProjectIdentifier: projectId,
+            name: schemaName,
+            typeIdentifier: 'MskTopicAssetType',
+            formsInput,
+            externalIdentifier: buildMskTopicArn(region, accountId, clusterName, schemaName, partition),
+          }));
+          
+          const newAssetId = createResponse.id;
+          
+          // Store the new asset ID in SSM Parameter Store
+          const putParameterCommand = new PutParameterCommand({
+            Name: parameterName,
+            Value: newAssetId,
+            Type: 'String',
+            Overwrite: true
+          });
+          
+          await ssmClient.send(putParameterCommand);
+          console.log(`Stored asset ID ${newAssetId} in SSM Parameter Store under ${parameterName}`);
+        } catch (error) {
+          console.error('Error creating MSK asset:', error);
+        }
+      }
+    }
+    
+    // Step 5: Clean up old assets if necessary
+    for (const [assetName, assetId] of assetMap) {
+      if (!schemas.some(schema => schema.SchemaName === assetName)) {
+        // If the schema is not in the current list, delete the asset and its parameter
+        console.log(`Deleting stale asset ${assetName}.`);
+        try {
+          await dataZoneClient.send(new DeleteAssetCommand({
+            domainIdentifier: domainId,
+            identifier: assetId
+          }));
+          
+          await ssmClient.send(new DeleteParameterCommand({ Name: `/datazone/${domainId}/${registryName}/asset/${assetName}` }));
+          console.log(`Deleted asset ${assetName} and its parameter.`);
+        } catch (error) {
+          console.error('Error deleting asset or parameter:', error);
+        }
+      }
+    }
+    
+  } catch (error) {
+    console.error('Error processing request:', error);
+    return {
+      statusCode: 500,
+      body: JSON.stringify({
+        message: 'Failed to process request',
+        error: error.message
+      }),
+    };
+  }
+};
+
+// Utility functions
+function buildMskTopicArn(region, accountId, clusterName, topicName, partition) {
+  return `arn:${partition}:kafka:${region}:${accountId}:topic/${clusterName}/${topicName}`;
```

Comment:
this is wrong, this is the actual ARN format:

`arn:${Partition}:kafka:${Region}:${Account}:topic/${ClusterName}/${ClusterUuid}/${TopicName}`